File: C++/Problem003.cc
// The prime factors of 13195 are 5, 7, 13 and 29.
// What is the largest prime factor of the number 600851475143 ?

#include <iostream>
#include <math.h>
#include "LinkedList.h"

class Prime {
    private:
        int64_t n;
        LinkedList<int> factor_list;

    public:
        Prime(int64_t query) {
            n = query;
        }

        void getFactors() {
            factor_list = LinkedList<int>();
            std::cout << n;
        }
};

main() {
    // int64_t query = 13195;
    // int64_t query = 600851475143;

    int64_t query = 98;

    Prime prime = Prime(query);
    prime.getFactors();

}

File: Java/Problem003.java
public class Problem003 implements Problem {

    //The prime factors of 13195 are 5, 7, 13 and 29.
    //What is the largest prime factor of the number 600851475143 ?

    public static void main(String[] args) {
        new Problem003().solve();
    }

    @Override
    public void solve() {
        long n = 600851475143L;
        int i = 2;

        while (n != 1) {
            while (i <= n) {
                if (n % i == 0) {
                    n /= i;
                    if (n == 1)
                        System.out.println("Largest prime: " + i);
                } else {
                    break;
                }
            }
            i++;
        }
    }
}
